packages/redux/src/sharedWishlists/middlewares/updateSharedWishlistUponItemsChanges.ts

```diff
@@ -0,0 +1,56 @@
+import * as actionTypes from '../../wishlists/actionTypes.js';
+import {
+  getSharedWishlistId,
+  type UpdateSharedWishlistAction,
+} from '../index.js';
+import { getUser, getUserIsGuest } from '../../users/selectors.js';
+import { updateSharedWishlist } from '../actions/index.js';
+import type {
+  AddWishlistItemAction,
+  GetOptionsArgument,
+  RemoveWishlistItemAction,
+  StoreState,
+  UpdateWishlistItemAction,
+} from '../../index.js';
+import type { Middleware } from 'redux';
+import type { ThunkDispatch } from 'redux-thunk';
+
+type UpdateSharedWishlistUponItemsChangesParams = {
+  // Redux action dispatch.
+  dispatch: ThunkDispatch<
+    StoreState,
+    GetOptionsArgument,
+    UpdateSharedWishlistAction
+  >;
+  // Returns the current redux state.
+  getState: () => StoreState;
+};
+
+const updateSharedWishlistUponItemsChanges: Middleware =
+  ({ dispatch, getState }: UpdateSharedWishlistUponItemsChangesParams) =>
+  next =>
+  (
+    action:
```

Comment:
`action` here can be any action, not only those related to wishlists. The correct type should be `AnyAction` from redux.

Reply:
Updated ✅